Core/src/Material/VolumeMaterialMapper.cpp

```diff
@@ -312,9 +313,7 @@ void Acts::VolumeMaterialMapper::mapMaterialTrack(
         float remainder = rmIter->materialProperties.thickness() -
                           m_cfg.mappingStep * volumeStep;
         properties.scaleThickness(m_cfg.mappingStep / properties.thickness());
-        mState.recordedMaterial[volIter->volume->geoID()].push_back(
-            std::pair(properties, rmIter->position));
-        for (int step = 1; step <= volumeStep; step++) {
+        for (int step = 0; step <= volumeStep; step++) {
```

Comment:
Hi, I think the initial protection against `step==0` is that your first step might not give you a direction ... can we make a more robust way for that?

Reply:
Here step==0 does not mean the first step, the step correspond to additional measurement I create between two Geantino interaction to have a more or less uniform sampling of the volume. So Step==0 correspond to the geantino interaction.
So here the direction is the direction of the geantino and there is no reason why it shouldn't have one (I think).
But maybe I should change change step to extrapolated measurement or something similar to make it clearer ?

Reply:
Ok, thanks for the clarification.

Reply:
Looking at it again I realise that the Direction shouldn't be computed inside the loop so I will change that quickly.